tests/unit/ui/components/TestIconStar.js

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+import Icon from 'ui/components/Icon';
+import IconStar, {
+  getSvgPath,
+  CLOSED_STYLE,
+  DIM_CLOSED_STYLE,
+  HALF_STYLE,
+  OPEN_STYLE,
+  YELLOW,
+  GRAY,
+} from 'ui/components/IconStar';
+
+describe(__filename, () => {
+  it('sets the default color to YELLOW', () => {
+    const star = shallow(<IconStar />);
+
+    expect(star.find('g')).toHaveProp('fill', YELLOW);
+  });
+
+  it('changes the color to GRAY if yellow is false', () => {
+    const star = shallow(<IconStar yellow={false} />);
+
+    expect(star.find('g')).toHaveProp('fill', GRAY);
+  });
+
+  it('sets the star style to HALF_STYLE if half and readOnly are true', () => {
+    const star = shallow(<IconStar readOnly half />);
+
+    expect(star.find('defs')).toHaveLength(1);
+    expect(star.find('path').html()).toContain(getSvgPath(HALF_STYLE));
+  });
+
+  it("sets the star style to CLOSED_STYLE if the star is selected and it's not readOnly", () => {
+    const star = shallow(<IconStar readOnly={false} selected />);
+
+    expect(star.find('g')).toHaveProp('fillOpacity', 1);
+    expect(star.find('path').html()).toContain(getSvgPath(CLOSED_STYLE));
```

Comment:
It's a bit better to use `toHaveProp` here because I think it will not do a full render as `html()` does (but I'm not totally sure if that's true):

```diff
diff --git a/tests/unit/ui/components/TestIconStar.js b/tests/unit/ui/components/TestIconStar.js
index 63fde7c22..17086dd2d 100644
--- a/tests/unit/ui/components/TestIconStar.js
+++ b/tests/unit/ui/components/TestIconStar.js
@@ -36,7 +36,7 @@ describe(__filename, () => {
     const star = shallow(<IconStar readOnly={false} selected />);
 
     expect(star.find('g')).toHaveProp('fillOpacity', 1);
-    expect(star.find('path').html()).toContain(getSvgPath(CLOSED_STYLE));
+    expect(star.find('path')).toHaveProp('d', getSvgPath(CLOSED_STYLE));
   });
 
   it("sets the star style to DIM_CLOSED_STYLE if the star is not selected and it's readOnly", () => {
```

Reply:
sure thing, I'll try it! thanks